qiskit/circuit/qpy_serialization.py

```diff
@@ -0,0 +1,706 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name,too-many-boolean-expressions,redundant-keyword-arg
+
+"""
+===========================================================
+QPY serialization (:mod:`qiskit.circuit.qpy_serialization`)
+===========================================================
+
+.. currentmodule:: qiskit.circuit.qpy_serialization
+
+.. autosummary::
+
+    load
+    dump
+
+QPY Format
+==========
+
+The QPY serialization format is a a portable cross-platform binary
+serialization format for QuantumCircuit objects in Qiskit. The basic
+format for the file format is as follows:
+
+A QPY file (or memory object) always starts with the following 7
+byte UTF8 string: ``QISKIT`` which is immediately followed by the overall
+file header. The contents of thie file header as defined as a C struct are:
+
+.. code-block:: c
+
+    struct {
+        unsigned char qpy_version;
+        unsigned char qiskit_major_version;
+        unsigned char qiskit_minor_version;
+        unsigned char qiskit_patch_version;
+        unsigned long long num_circuits;
+    }
+
+All values use network byte order [#f1]_ (big endian) for cross platform
+compatibility. All strings will be padded with ``0x00`` after the string
+if the string is shorter than the max size.
```

Comment:
I removed this sentence in https://github.com/Qiskit/qiskit-terra/pull/5578/commits/77b995e2aa54e6909a598a3dbf7e2bb2d80f7ce4 since I've moved away from fixed width strings.